xdsl/ir/core.py

```diff
@@ -55,28 +56,53 @@ def attributes(self) -> Iterator[type[Attribute]]:
 class MLContext:
     """Contains structures for operations/attributes registration."""
 
-    _registeredOps: dict[str, type[Operation]] = field(default_factory=dict)
-    _registeredAttrs: dict[str, type[Attribute]] = field(default_factory=dict)
+    _loaded_ops: dict[str, type[Operation]] = field(default_factory=dict)
```

Comment:
I'm surprised we need to change MLContext at all. Couldn't we get away with changes only in xdsl-opt?

Reply:
For example, with a set of dialect dependencies per pass.

Reply:
That would be annoying to do, because some passes are purely generic (like `dce`).
Also, the `MLContext` is really only used in the parser, so I guess that's fine.

To be honest, I was happy that this was the only change that was required in the core code ^^